python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -198,14 +198,24 @@ def run(self, exposure, templateExposure, subtractedExposure, psfMatchingKernel,
 
         if svar is None:
             svar = self.computeVarianceMean(exposure)
+        if np.isnan(svar):  # Should not happen unless entire image has been masked.
+            svar = 1e-9
```

Comment:
I understand your point. The cases where I have had this happen is when using the ImageMapReduce framework (which does these computations on a grid of subimages) when a subimage is covered by a mask. Now that you mention it, it can be short-circuited and just return a copy of the input subtractedExposure, which is all NaNs and/or masked. I'll do that.